docs/dev-guide/policies/style.rst

```diff
@@ -11,11 +11,33 @@ length limit is 100 characters, which we chose to prevent line wrapping on GitHu
 In-code Documentation
 ---------------------
 
-Document your functions using the markup described
-`here <http://sphinx-doc.org/markup/desc.html#info-field-lists>`_.
-If it's worth having a function, it's worth taking 1 minute to describe what it
-does, define each parameter, and define its return value. This saves a
-tremendous amount of time for the next person who looks at your code.
+All new code must have a doc block that describes what the function or class does,
+describes each of the parameters, lists the parameter types (fully qualified please), lists
```

Comment:
what about:
"describes each of the parameters and its type (fully qualified please), lists
exceptions that can be raised, describes the return value and its type.